src/Compilers/CSharp/Portable/Symbols/FunctionPointers/FunctionPointerMethodSymbol.cs

```diff
@@ -98,6 +93,68 @@ public static FunctionPointerMethodSymbol CreateFromSource(FunctionPointerTypeSy
                 diagnostics,
                 suppressUseSiteDiagnostics);
+
+            // PROTOTYPE(func-ptr): Do this right. Just making compatible changes for now
+            static CallingConvention getCallingConvention(FunctionPointerCallingConventionSyntax? callingConventionSyntax, DiagnosticBag diagnostics)
+            {
+                switch (callingConventionSyntax?.ManagedOrUnmanagedKeyword.Kind())
+                {
+                    case null:
+                        return CallingConvention.Default;
+
+                    case SyntaxKind.ManagedKeyword:
+                        // Possible if we get a node not constructed by the parser
+                        if (callingConventionSyntax.UnmanagedCallingConventionList is object && !callingConventionSyntax.ContainsDiagnostics)
+                        {
+                            diagnostics.Add(ErrorCode.ERR_CannotSpecifyManagedWithUnmanagedSpecifiers, callingConventionSyntax.UnmanagedCallingConventionList.GetLocation());
+                        }
+                        return CallingConvention.Default;
+
+                    case SyntaxKind.UnmanagedKeyword:
+                        switch (callingConventionSyntax.UnmanagedCallingConventionList)
+                        {
+                            case null:
+                                return CallingConvention.Unmanaged;
+
+                            case { CallingConventions: { Count: 1 } specifiers }:
+                                return specifiers[0].Name switch
+                                {
+                                    { Text: "Cdecl" } => CallingConvention.CDecl,
+                                    { Text: "Stdcall" } => CallingConvention.Standard,
+                                    { Text: "Thiscall" } => CallingConvention.ThisCall,
+                                    { Text: "Fastcall" } => CallingConvention.FastCall,
+                                    // PROTOTYPE(func-ptr): Handle unrecognized specifiers
+                                    var name => reportBadConventionAndReturn(name.Text, name, diagnostics)
+                                };
+
+                            case { CallingConventions: { Count: 0 } } unmanagedList:
+                                // Should never be possible from parser-constructed code (parser will always provide at least a missing identifier token),
+                                // so diagnostic quality isn't hugely important
+                                return reportBadConventionAndReturn(givenConvention: "", unmanagedList, diagnostics);
+
+                            case { CallingConventions: var specifiers }:
+                                // PROTOTYPE(func-ptr): Handle multiple specifiers
+                                foreach (var convention in specifiers)
+                                {
+                                    _ = reportBadConventionAndReturn(convention.Name.Text, convention.Name, diagnostics);
```

Comment:
>reportBadConventionAndReturn(convention.Name.Text, convention.Name, diagnostics) [](start = 40, length = 80)

I guess this diagnostics could be misleading because the identifiers could be from the list the messages suggests as acceptable. Probably fine as a temporary solution until we get to addressing the PROTOTYPE comment

Reply:
Yeah, my intention is that much of this method will have to be rewritten, and that we'll only flag actually invalid things. I wanted to do the simplest possible thing for now.

---
In reply to: [452354819](https://github.com/dotnet/roslyn/pull/45684#discussion_r452354819) [](ancestors = 452354819)